### PR TITLE
Changed libraries to start using Java 17 and not Kotlin

### PR DESCRIPTION
**Pull request by mnlx**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+##ktorVersion=2.3.2
+#logbackVersion=1.4.8
+##kotlinxSerializationVersion=1.5.1
+##kotlinOpenaiVersion=3.3.0
+##dotenvKotlinVersion=6.4.1
+##kotlin.code.style=official
+#org.gradle.parallel=true
+#org.gradle.jvmargs=-Xmx3g -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
+#
+#openAiServiceVersion=0.15.0
```
Comment (juanjoman) [inline gradle.properties]: can we remove the kotlin properties? Why are the rest commented?

File: modules/embeddingsgeneration/build.gradle [hunk 2]
```diff
@@ -0,0 +1,17 @@
+plugins {
+    id 'java'
+}
+
+group = "com.theagilemonkeys.ellmental.modules.embeddingsmodel"
+version = 'unspecified'
```
Comment (juanjoman) [inline modules/embeddingsgeneration/build.gradle]: I think we can set 0.0.1-SNAPSHOT for now 😄

File: modules/embeddingsgeneration/src/main/java/com/theagilemonkeys/ellmental/embeddingsgeneration/Embedding.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package com.theagilemonkeys.ellmental.embeddingsgeneration;
+
+public class Embedding {
+}
```
Comment (juanjoman) [inline modules/embeddingsgeneration/src/main/java/com/theagilemonkeys/ellmental/embeddingsgeneration/Embedding.java]: Why is the embedding empty? 😮

File: modules/embeddingsstore/build.gradle [hunk 4]
```diff
@@ -0,0 +1,19 @@
+plugins {
+    id 'java'
+}
+
+group = 'org.example'
+version = 'unspecified'
```
Comment (juanjoman) [inline modules/embeddingsstore/build.gradle]: same about the version

File: modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/Embedding.java [hunk 5]
```diff
@@ -0,0 +1,4 @@
+package com.theagilemonkeys.ellmental.embeddingsstore;
+
+public class Embedding {
+}
```
Comment (juanjoman) [inline modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/Embedding.java]: same for the empty object

**Review by juanjoman — CHANGES_REQUESTED** [review]
Just added a couple of suggestions and questions, nice work!!

File: modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/Embedding.java (hunk 5 above)
Comment (mnlx) [reply modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/Embedding.java]: These where just the scafolding for the initial setup. At first I just wanted to validate if the project structure was as expected. I just pushed the updates on these with their non empty versions.

File: modules/embeddingsgeneration/src/main/java/com/theagilemonkeys/ellmental/embeddingsgeneration/Embedding.java (hunk 3 above)
Comment (mnlx) [reply modules/embeddingsgeneration/src/main/java/com/theagilemonkeys/ellmental/embeddingsgeneration/Embedding.java]: Answered below

File: modules/embeddingsgeneration/build.gradle (hunk 2 above)
Comment (mnlx) [reply modules/embeddingsgeneration/build.gradle]: Updated with the latest push

File: gradle.properties (hunk 1 above)
Comment (mnlx) [reply gradle.properties]: I'll remove them in the next push

File: modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/pinecone/PineconeEmbeddingsStore.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        MediaType JSON
+                = MediaType.parse("application/json; charset=utf-8");
+
+        RequestBody body = RequestBody.create(bodyString, JSON);
+
+        Request request = new Request.Builder()
+                .url(url + path)
+                .header("accept", "application/json")
+                .header("content-type", "application/json")
+                .header("Api-Key", "1c13987f-1d13-4373-8655-f739089de6af")
```
Comment (mnlx) [inline modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/pinecone/PineconeEmbeddingsStore.java]: This API key is from my personal account, I'll delete them once I implement dotenv or env variables.

File: modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/pinecone/UpsertVectorSchema.java [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+
+
+
+
+
+
+
+
+//class Schema {
```
Comment (mnlx) [inline modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/pinecone/UpsertVectorSchema.java]: Removing this

File: modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/pinecone/UpsertVectorSchema.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    UpsertVectorSchema(Embedding embedding, Map<String,String> metadata ) {
+        vectors = Arrays.asList(new Vector(UUID.randomUUID().toString(), embedding.vector, metadata));
+    }
+
+}
+class SparseValues {
+    public List<Integer> indices;
+    public List<Integer> values;
+
+    // Getters and setters
```
Comment (juanjoman) [inline modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/pinecone/UpsertVectorSchema.java]: You can have a look at lombok if you want to get rid of a lot of code 😋 -> https://www.baeldung.com/intro-to-project-lombok

**Review by NickSeagull — APPROVED** [review]
Apart of deleting dead code or env keys, lgtm!

**Review by adrian-lorenzo — APPROVED** [review]
The same as Nick. Apart from that, LGTM!

File: modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/pinecone/UpsertVectorSchema.java (hunk 8 above)
Comment (mnlx) [reply modules/embeddingsstore/src/main/java/com/theagilemonkeys/ellmental/embeddingsstore/pinecone/UpsertVectorSchema.java]: I heard of this library before, never new it was used for this. So much easier. I'm going to create a task to start using Lombok.

**Pull request merged**